### PR TITLE
ELPP-3588 Update PHP image and use Docker Compose

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Based on elifesciences/api-dummy#131

### CODE REVIEW

File: Dockerfile.composer [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+ARG composer_dev_arg
+
+COPY composer.json \
+    composer.lock \
+    ./
+
+RUN composer --no-interaction install ${composer_dev_arg} --ignore-platform-reqs --no-autoloader --no-scripts --no-suggest
+
+COPY test/ test/
```
Comment (giorgiosironi) [inline Dockerfile.composer]: There is a `src/` here, but it doesn't have any classes; it could still be put in for simplicity

File: Jenkinsfile [hunk 2]
```diff
@@ -7,13 +7,18 @@ elifeLibrary {
...
     stage 'Build image', {
-        dockerBuild 'hypothesis-dummy', commit
+        sh "IMAGE_TAG=${commit} docker-compose -f docker-compose.yml -f docker-compose.ci.yml build"
         image = DockerImage.elifesciences(this, 'hypothesis-dummy', commit)
     }
 
     stage 'Project tests', {
-        dockerBuildCi 'hypothesis-dummy', commit
         dockerProjectTests 'hypothesis-dummy', commit
+        try {
```
Comment (giorgiosironi) [inline Jenkinsfile]: I'd be extracting this later

File: Dockerfile.composer (hunk 1 above)
Comment (thewilkybarkid) [reply Dockerfile.composer]: Things are only here for the autoloader. Not having it saves superfluous cache busting (and future classes not appearing in the autoloader will be an obvious fail).

Comment (thewilkybarkid) [reply Dockerfile.composer]: Ideally `test` wouldn't be copied if `${composer_dev_arg}` is empty.

**Pull request merged**